common/djangoapps/student/views/management.py

```diff
@@ -893,18 +898,26 @@ def confirm_email_change(request, key):
         return response
 
 
-@require_POST
-@login_required
-@ensure_csrf_cookie
+@api_view(['POST'])
+@authentication_classes((
+    JwtAuthentication,
+    BearerAuthenticationAllowInactiveUser,
+    SessionAuthenticationAllowInactiveUser,
+))
+@permission_classes((IsAuthenticated,))
 def change_email_settings(request):
     """
     Modify logged-in user's setting for receiving emails from a course.
     """
     user = request.user
 
     course_id = request.POST.get("course_id")
-    course_key = CourseKey.from_string(course_id)
     receive_emails = request.POST.get("receive_emails")
+    if course_id is None:
```

Comment:
we should use `request.data` instead and `request.POST` and remove this `if` block

Reply:
Let me confirm it works with request.data in both ways. I found that while making request from postman it was coming in request.POST and while running through edx api client it was coming in request.data.